File: app/src/main/java/kr/co/jayhy/weather/repo/WeatherForecastRepo.java
package kr.co.jayhy.weather.repo;

import android.support.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

/**
 * Created by jayhy on 2017. 8. 12..
 */

public class WeatherForecastRepo {

    @SerializedName("cod")
    public String cod;

    @SerializedName("city")
    public City city;

    @SerializedName("cnt")
    public String cnt;

    @SerializedName("list")
    public ArrayList<List> list;

    public class City {

        @SerializedName("id")
        public String id;

        @SerializedName("name")
        public String name;

        @SerializedName("coord")
        public Coord coord;

        @SerializedName("country")
        public String country;

        public class Coord {
            @SerializedName("lon")
            public String lon;

            @SerializedName("lat")
            public String lat;

        }
    }


    public class List {
        @SerializedName("dt")
        public String dt;

        @SerializedName("main")
        public Main main;

        @SerializedName("weather")
        public ArrayList<Weather> weathers;

        @SerializedName("clouds")
        public Clouds clouds;

        @SerializedName("wind")
        public Wind wind;

        @SerializedName("rain")
        public Rain rain;

        @SerializedName("snow")
        public Snow snow;

        @SerializedName("dt_txt")
        public String dt_txt;

        public class Main {
            @SerializedName("temp")
            public String temp;

            @SerializedName("temp_min")
            public String temp_min;

            @SerializedName("temp_max")
            public String temp_max;

            @SerializedName("pressure")
            public String pressure;

            @SerializedName("sea_level")
            public String sea_level;

            @SerializedName("grnd_level")
            public String grnd_level;

            @SerializedName("humidity")
            public String humidity;

            @SerializedName("temp_kf")
            public String temp_kf;

        }

        public class Weather {
            @SerializedName("id")
            public String id;

            @SerializedName("main")
            public String main;

            @SerializedName("description")
            public String description;

            @SerializedName("icon")
            public String icon;

        }

        public class Clouds {
            @SerializedName("all")
            public String all;
        }

        public class Wind {
            @SerializedName("speed")
            public String speed;

            @SerializedName("deg")
            public String deg;

        }

        public class Rain {
            @Nullable
            @SerializedName("3h")
            public String rain;
        }

        public class Snow {
            @Nullable
            @SerializedName("3h")
            public String snow;
        }

    }


}

File: kotlin/src/main/java/kr/co/jayhy/weather/repo/WeatherForecastRepo.kt
package kr.co.jayhy.weather.repo

import com.google.gson.annotations.SerializedName

import java.util.ArrayList

/**
 * Created by jayhy on 2017. 8. 12..
 */

class WeatherForecastRepo {

    @SerializedName("cod")
    var cod: String? = null

    @SerializedName("city")
    var city: City? = null

    @SerializedName("cnt")
    var cnt: String? = null

    @SerializedName("list")
    var list: ArrayList<List>? = null

    inner class City {

        @SerializedName("id")
        var id: String? = null

        @SerializedName("name")
        var name: String? = null

        @SerializedName("coord")
        var coord: Coord? = null

        @SerializedName("country")
        var country: String? = null

        inner class Coord {
            @SerializedName("lon")
            var lon: String? = null

            @SerializedName("lat")
            var lat: String? = null

        }
    }


    inner class List {
        @SerializedName("dt")
        var dt: String? = null

        @SerializedName("main")
        var main: Main? = null

        @SerializedName("weather")
        var weathers: ArrayList<Weather>? = null

        @SerializedName("clouds")
        var clouds: Clouds? = null

        @SerializedName("wind")
        var wind: Wind? = null

        @SerializedName("rain")
        var rain: Rain? = null

        @SerializedName("snow")
        var snow: Snow? = null

        @SerializedName("dt_txt")
        var dt_txt: String? = null

        inner class Main {
            @SerializedName("temp")
            var temp: String? = null

            @SerializedName("temp_min")
            var temp_min: String? = null

            @SerializedName("temp_max")
            var temp_max: String? = null

            @SerializedName("pressure")
            var pressure: String? = null

            @SerializedName("sea_level")
            var sea_level: String? = null

            @SerializedName("grnd_level")
            var grnd_level: String? = null

            @SerializedName("humidity")
            var humidity: String? = null

            @SerializedName("temp_kf")
            var temp_kf: String? = null

        }

        inner class Weather {
            @SerializedName("id")
            var id: String? = null

            @SerializedName("main")
            var main: String? = null

            @SerializedName("description")
            var description: String? = null

            @SerializedName("icon")
            var icon: String? = null

        }

        inner class Clouds {
            @SerializedName("all")
            var all: String? = null
        }

        inner class Wind {
            @SerializedName("speed")
            var speed: String? = null

            @SerializedName("deg")
            var deg: String? = null

        }

        inner class Rain {
            @SerializedName("3h")
            var rain: String? = null
        }

        inner class Snow {
            @SerializedName("3h")
            var snow: String? = null
        }

    }


}
